### PR TITLE
Feature/card errors

### PR DESCRIPTION
**Pull request by devights**

### CODE REVIEW

File: myuw/templates/handlebars/card/calendars/acad_cal_sp.html [hunk 1]
```diff
@@ -13,6 +13,9 @@ <h2 class="myuw-card-title">Academic Calendar</h2>
 	        {{ /each }}
 	    </ul>
    </div>
+    {{else}}
+    <p>ERROR</p>
```
Comment (fanglinfang) [inline myuw/templates/handlebars/card/calendars/acad_cal_sp.html]: Is the ERROR here means actual message?

File: myuw/templates/handlebars/card/instructor_schedule/course_eval.html [hunk 2]
```diff
@@ -82,7 +82,7 @@ <h4 class="myuw-card-row-heading">Evaluations</h4>
                       </li>
                   {{else}}
                       <li class="evaluation-schedule">
-                          <span class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error has occurred and we can't load this content right now. Please try again later.</span>
+                          <span class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred with the <a href="http://www.washington.edu/assessment/course-evaluations/" data-linklabel="Course Evaluation" style="text-decoration:underline;" target="_blank">Course evaluation</a>. Please try again later.</span>
```
Comment (fanglinfang) [inline myuw/templates/handlebars/card/instructor_schedule/course_eval.html]: The course eval sites are campus & program specific. This link might not be appropriate.

File: myuw/views/api/instructor_schedule.py [hunk 3]
```diff
@@ -108,6 +108,7 @@ def set_section_grading_status(section, person):
         else:
             raise
     except Exception:
+        return "error"
```
Comment (fanglinfang) [inline myuw/views/api/instructor_schedule.py]: Do you really mean to return before log_exception?

Comment (devights) [reply myuw/views/api/instructor_schedule.py]: Good catch, fixed

File: myuw/templates/handlebars/card/calendars/acad_cal_sp.html (hunk 1 above)
Comment (hlee2) [reply myuw/templates/handlebars/card/calendars/acad_cal_sp.html]: Message added.

File: myuw/templates/handlebars/card/instructor_schedule/course_eval.html (hunk 2 above)
Comment (hlee2) [reply myuw/templates/handlebars/card/instructor_schedule/course_eval.html]: Hyperlink is removed

File: myuw/templates/handlebars/card/accounts/library.html [hunk 4]
```diff
@@ -59,6 +59,8 @@ <h4 class="card-badge-label">You owe</h4>
                 <span class="btn btn-default card-action-btn pull-right"><a href="http://search.lib.uw.edu/account" target="_blank" data-linklabel="Your Library Account">Access library account</a></span>
             </div>
         </div>
-
+    {{else}}
+    <div><p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your library account information right now. In the meantime, try the <a href="http://www.lib.washington.edu/" data-linklabel="UW Libraries" style="text-decoration:underline;" target="_blank">UW Libraries page</a>.</p></div>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/accounts/library.html]: I recommend removing the inline styles here for underlining and adding that style to the CSS.

File: myuw/templates/handlebars/card/accounts/tuition.html [hunk 5]
```diff
@@ -173,7 +173,12 @@ <h4>Financial Aid</h4>
         <!-- finaid notices end -->
 		{{/if}}
      </div>
-
+    {{else}}
+        <div><p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your information right now. In the meantime, try the <a href="https://sdb.admin.uw.edu/sisStudents/uwnetid/tuition.aspx" data-linklabel="Tuition Statement" style="text-decoration:underline;" target="_blank">Tuition Statement page</a>.</p></div>
+        {{#if is_pce}}
+        <div><p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your information right now. In the meantime, try the <a href="https://portal.continuum.uw.edu" data-linklabel="PCE Tuition portal" style="text-decoration:underline;" target="_blank">PCE Tuition portal</a>.</p></div>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/accounts/tuition.html]: Same with these inline styles.

File: myuw/templates/handlebars/card/accounts/upass.html [hunk 6]
```diff
@@ -114,6 +114,9 @@ <h4>Summer U-PASS Use</h4>
             {{/if}}
         {{/if}}
     {{/if}}
+{{else}}
+<div><p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your information right now. In the meantime, try the <a href="http://facilities.uw.edu/transportation/index" data-linklabel="UW Transportation" style="text-decoration:underline;" target="_blank">UW Transportation page</a>.</p></div>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/accounts/upass.html]: And here

File: myuw/templates/handlebars/card/accounts/uwnetid.html [hunk 7]
```diff
@@ -18,5 +19,8 @@ <h3>UW NetID </h3>
 	    {{/if}}
 	</ul>
     </div>
+    {{else }}
+    <div><p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error has occurred and MyUW cannot load your information right now. In the meantime, if you want to change your password, try the <a href="https://uwnetid.washington.edu/manage/?password" data-linklabel="UW NetID page" style="text-decoration:underline;" target="_blank">UW NetID page</a>.</p></div>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/accounts/uwnetid.html]: and here

File: myuw/templates/handlebars/card/grad_committee.html [hunk 8]
```diff
@@ -24,8 +25,10 @@ <h4>{{committee_type}}</h4>
         </div>
         <div class="clearfix">
             <div class="card-badge-action pull-right">
-                <a href="http://www.grad.uw.edu/mygrad/student.htm" data-linklabel="MyGrad">Go to MyGrad</a>
+                <a href="http://www.grad.uw.edu/mygrad/student.htm" data-linklabel="MyGrad" target="_blank">Go to MyGrad</a>
             </div>
-        </div>
+        {{else}}
+        <p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i>An error occurred and MyUW cannot load your committees information right now. In the meantime, try the <a href="http://grad.uw.edu/for-students-and-post-docs/mygrad-program/" data-linklabel="MyGrad" style="text-decoration:underline;" target="_blank">MyGrad program page</a>.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/grad_committee.html]: and here

File: myuw/templates/handlebars/card/grad_status.html [hunk 9]
```diff
@@ -107,8 +107,11 @@ <h5>{{req_type}}, {{target_award_quarter}} {{target_award_year}}</h5>
 
         <div class="clearfix">
             <div class="card-badge-action pull-right">
-                <a href="http://www.grad.uw.edu/mygrad/student.htm" data-linklabel="MyGrad">Go to MyGrad</a>
+                <a href="http://www.grad.uw.edu/mygrad/student.htm" data-linklabel="MyGrad" target="_blank">Go to MyGrad</a>
             </div>
         </div>
+        {{else}}
+        <p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your graduate request information right now. In the meantime, try the <a href="http://grad.uw.edu/for-students-and-post-docs/mygrad-program/" data-linklabel="MyGrad" style="text-decoration:underline;" target="_blank">MyGrad program page</a>.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/grad_status.html]: and here

File: myuw/templates/handlebars/card/instructor_schedule/course_grading.html [hunk 10]
```diff
@@ -73,7 +73,9 @@ <h4 class="myuw-card-row-heading">Grade Submission</h4>
                                 {{/if}}
                             {{/if}}
                             <a href="https://itconnect.uw.edu/learn/tools/gradepage/" rel="{{curriculum_abbr}} {{course_number}} {{section_id}}" class="gradepage_help_link" target="_blank" data-linklabel="GradePage Help"><i class="fa fa-question-circle" aria-hidden="true"></i></a>
-                            {{/if}}
+                            {{else}}
+                            <p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred with <a href="https://gradepage.uw.edu/" data-linklabel="Gradepage" style="text-decoration:underline;" target="_blank">Gradepage</a>. Please try again later.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/instructor_schedule/course_grading.html]: and here

File: myuw/templates/handlebars/card/instructor_schedule/course_resource_panel.html [hunk 11]
```diff
@@ -36,6 +36,8 @@ <h4 class="myuw-card-row-heading">Course Materials</h4>
                             <li class="clearfix">
                                 {% include "handlebars/card/instructor_schedule/course_resource/email_list.html" %}
                             </li>
+                        {{else}}
+                        <p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred with the <a href="https://mailman.u.washington.edu/" data-linklabel="UW Mailman" style="text-decoration:underline;" target="_blank">email list</a>. Please try again later.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/instructor_schedule/course_resource_panel.html]: and here.

File: myuw/templates/handlebars/card/profile/employee_profile.html [hunk 12]
```diff
@@ -105,5 +106,9 @@ <h4 class="myuw-card-row-heading">UW Directory</h4>
             </div>
         </div>
     </div>
+    {{else}}
+    <!-- WIP: Employee profile error -->
+    <div><p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error has occurred and MyUW cannot load your profile information right now. In the meantime, try <a href="https://wd5.myworkday.com/uw/d/home.htmld" data-linklabel="Workday" style="text-decoration:underline;" target="_blank">Workday</a> if you need to change your address or the <a href="http://www.washington.edu/home/peopledir/" data-linklabel="UW Directory" style="text-decoration:underline;" target="_blank">UW directory</a> if you need to change your directory settings.</p></div>
+    {{/unless}}
```
Comment (klpittman) [inline myuw/templates/handlebars/card/profile/employee_profile.html]: and here

File: myuw/templates/handlebars/card/reg_status.html [hunk 13]
```diff
@@ -80,8 +80,12 @@ <h4 class="card-badge-label">{{ pluralize_by_size ../pending_minors "Minor" "Min
             {{#if plan_data}}
                 {{> myplan_courses }}
             {{/if}}
+
+         <!-- WIP: Regi card failure inline message -->
+		 <p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred, please try again later. If you are attempting to register, you can try this <a href="https://sdb.admin.uw.edu/students/uwnetid/register.asp" data-linklabel="Registration using SLN" style="text-decoration:underline;" target="_blank">SLN registration page</a>.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/reg_status.html]: and here

File: myuw/templates/handlebars/card/registration/in_myplan.html [hunk 14]
```diff
@@ -87,5 +87,8 @@ <h4 class="card-badge-label">In MyPlan</span></h4>
         </div>
     </div>
   {{/if}}
+
+ <!-- WIP: MyPlan inline error message -->
+<p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your information right now. Please try <a href="https://uwstudent.washington.edu/student/myplan/mplogin/netid?rd=/student/myplan/" data-linklabel="MyPlan" style="text-decoration:underline;" target="_blank">MyPlan</a>.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/registration/in_myplan.html]: and here

File: myuw/templates/handlebars/card/schedule/course_cards.html [hunk 15]
```diff
@@ -42,4 +42,8 @@ <h4>
     </div>
     {{/each}}
 </div>
+{{else}}
+<p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> Some course links are not available right now. In the meantime, try  
+ <a href="https://canvas.uw.edu/" data-linklabel="Canvas LMS" style="text-decoration:underline;" target="_blank">Canvas LMS </a> or <a href="https://catalyst.uw.edu/" data-linklabel="Catalyst" style="text-decoration:underline;" target="_blank">Catalyst Web Tools</a>.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/schedule/course_cards.html]: in these two cases.

File: myuw/templates/handlebars/card/summary/schedule.html [hunk 16]
```diff
@@ -63,6 +65,10 @@ <h4 class="myuw-card-heading">
                    <a href="/academic_calendar/#{{ year }},{{capitalizeString quarter }}">View {{capitalizeString quarter}} {{year}} important dates and deadlines.</a>
                </div>
             {{/unless}}
+             {{ else }}
+             <p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your teaching schedule right now. In the meantime, try the 
+ <a href="https://sdb.admin.uw.edu/sisMyUWClass/uwnetid/default.aspx" data-linklabel="MyClass" style="text-decoration:underline;" target="_blank">My Class Instructor Resources</a> page.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/summary/schedule.html]: and here

File: myuw/templates/handlebars/card/accounts/husky.html [hunk 17]
```diff
@@ -76,5 +77,10 @@ <h4 class="sr-only">Employee Husky Account</h4>
         {{/if}}
 		{{/if}}<!-- End of Tacoma affil    -->
     </div>
+    {{else}}
+    <div>
+	    <p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred and MyUW cannot load your Husky card balance right now. In the meantime, if you want to add funds, try the <a href="https://hfs.uw.edu/olco/Secure/AccountSummary.aspx" data-linklabel="Husky account information" style="text-decoration:underline;" target="_blank">UW HFS page</a>.</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/accounts/husky.html]: and here

File: myuw/templates/handlebars/card/accounts/account_medicine.html [hunk 18]
```diff
@@ -27,6 +28,9 @@ <h4 class="pw-exp-date">Password expiration</h4>
             <div class="pw-link"><a href="https://services.uwmedicine.org/passwordportal/login.htm">Change UW Medicine password</a></div>
         </div><!-- .card-badge -->
     {{/if}}
+    {{else}}
+    <div><p class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i>An error occurred and MyUW cannot load your information right now. In the meantime, try the <a href="https://services.uwmedicine.org/passwordportal/login.htm" data-linklabel="UW Medicine account site" style="text-decoration:underline;" target="_blank">UW Medicine Account page</a>.</p></div>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/accounts/account_medicine.html]: and here

File: myuw/templates/handlebars/card/instructor_schedule/course_eval.html [hunk 19]
```diff
@@ -82,7 +82,7 @@ <h4 class="myuw-card-row-heading">Evaluations</h4>
                       </li>
                   {{else}}
                       <li class="evaluation-schedule">
-                          <span class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error has occurred and we can't load this content right now. Please try again later.</span>
+                          <span class="text-danger"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> An error occurred with the Course evaluation. Please try again later.</span>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/instructor_schedule/course_eval.html]: I'm not sure "Course" needs to be capitalized here.

File: myuw/templates/handlebars/card/profile/profile_card.html [hunk 20]
```diff
@@ -1,10 +1,17 @@
...
+	<div>
+	<h2 class="heading-profile">
+	        {{#if has_preferred}}{{display_name}} <span>({{full_name}})</span>&nbsp;<span><a href="https://identity.uw.edu/" title="Manage your preferred name" data-linklabel="Manage preferred name - Identity.UW"><i class="fa fa-pencil" aria-hidden="true"></i><span class="sr-only">Manage your preferred name</span></a></span></h2>
+	        {{else}}
+	        {{full_name}}&nbsp;<span><a href="https://identity.uw.edu/" title="Manage your preferred name" data-linklabel="Manage preferred name - Identity.UW"><span class="sr-only">Manage your preferred name</span><i class="fa fa-pencil" aria-hidden="true"></i></a></span></h2>
+	        {{/if}}
+	</div>
+	{{else}}
+    <!-- WIP:Profile Name error -->
+	<p class="text-danger" style="margin-bottom:0.5em;"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> Profile -- Error loading content</p>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/profile/profile_card.html]: > Profile -- Error loading content

This error message seems a little different than all the others. Is this meant to be the final text?

Comment (hlee2) [reply myuw/templates/handlebars/card/profile/profile_card.html]: yes, it was the final copy.

File: myuw/templates/handlebars/card/schedule/course_cards.html (hunk 15 above)
Comment (klpittman) [inline myuw/templates/handlebars/card/schedule/course_cards.html]: Should we still link to Catalyst here? Isn't it mostly disabled for participants?

File: myuw/templates/handlebars/card/instructor_schedule/course_eval.html (hunk 19 above)
Comment (hlee2) [reply myuw/templates/handlebars/card/instructor_schedule/course_eval.html]: Good catch. Will change it to the lowercase c.

File: myuw/templates/handlebars/card/accounts/library.html (hunk 4 above)
Comment (hlee2) [reply myuw/templates/handlebars/card/accounts/library.html]: removed

File: myuw/templates/handlebars/card/accounts/tuition.html (hunk 5 above)
Comment (hlee2) [reply myuw/templates/handlebars/card/accounts/tuition.html]: removed

File: myuw/templates/handlebars/card/calendars/acad_cal_sp.html [hunk 21]
```diff
@@ -13,9 +13,9 @@ <h2 class="myuw-card-title">Academic Calendar</h2>
 	        {{ /each }}
 	    </ul>
    </div>
-    {{else}}
+	<div class="view-all"><a href="/academic_calendar/" title="Navigate to Notices">View all events</a></div>
```
Comment (klpittman) [inline myuw/templates/handlebars/card/calendars/acad_cal_sp.html]: Should the title here say something like "Navigate to the Calendar"? "Notices" seems weird here, but maybe I'm missing something.

Comment (hlee2) [reply myuw/templates/handlebars/card/calendars/acad_cal_sp.html]: Good catch. the Link title and the message are updated.

**Pull request merged**